DecipherNow/gm-fabric-dashboard
# Rev Deps and Replace JSON Inspector

This PR updates the project's dependencies to latest, including updating React and React-DOM to 16.0.0-beta-5. 

This is a significant update because React 16.x is the new "React Fiber" release with a compatibility shim that deprecated several APIs. The only component that was incompatible with React 16.x was JSON-Inspector, which we already wanted to replace due to sluggish performance.

This PR creates a replacement Inspector component that maintains enough of the CSS class name structure to allow @shanberg's styles in`jsonInspector.scss`to keep working. The search bar now filters our rows instead of highlighting matching terms, and the search and selection handlers are now router-driven and tied into browser history. 

Note: There is dead code in `jsoninspector.scss.` This is intentional and will be cleaned up in a subsequent PR by @shanberg in conjunction with style improvements for the new inspector component.

To test:

1. Checkout the `rev-deps` branch
2. Run `npm install`. You will likely see some warnings around some peer dependencies for React and React-DOM not being met. This is expected and shouldn't stop you from continuing.
3. Run `npm run test` (hitting A to run all **four** tests) and `npm run lint-js` to make sure you don't see any linting errors (if you see linting errors, something might be misconfigured in this project's `eslintrc` config, so let me know in comments below.
4. Open a second terminal tab and run `npm run mock-jvm.` This runs a mock backend that serves out fake data
5. Return to your original terminal tab and run `npm start`.
6. If you don't see all of the expected dashboards (JVM, HTTP, Finagle) or if you encounter issues in the UI, go to settings and Reset Dashboards to clear state stored in your local cache.
7. Navigate to the Explorer Component and search for 'GET'. The component should filter out all other metrics that don't contain this keyword. Look for the query to appear in your URL bar
8. Select a metric. This should render a chart and also appear in your URL bar.
9. Trigger a hard refresh of the browser. The UI should load and your search query and selected metric should be restored.

Resolves #276 
Resolves #277 
Resolves #278
